{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"></ul></div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "ntqtgvwte3xe",
    "outputId": "65ae5d17-f758-4093-896b-7182278d1ad9"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'kraken2'...\n",
      "remote: Enumerating objects: 120, done.\u001b[K\n",
      "remote: Counting objects: 100% (120/120), done.\u001b[K\n",
      "remote: Compressing objects: 100% (81/81), done.\u001b[K\n",
      "remote: Total 816 (delta 72), reused 73 (delta 39), pack-reused 696\u001b[K\n",
      "Receiving objects: 100% (816/816), 394.22 KiB | 2.07 MiB/s, done.\n",
      "Resolving deltas: 100% (561/561), done.\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/DerrickWood/kraken2.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "G9ZQscW7fDbj",
    "outputId": "f350e7ee-5d54-434e-e176-7b706b54a135"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/content/kraken2\n",
      "CHANGELOG.md\tdata  install_kraken2.sh  README.md  src\n",
      "CMakeLists.txt\tdocs  LICENSE\t\t  scripts\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "7a_19yL6fEx0",
    "outputId": "3acf976e-1cbb-4fe3-bc4b-b29bc67129c7"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'kraken2/'\n",
      "/content/kraken2\n",
      "make: Entering directory '/content/kraken2/src'\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o estimate_capacity.o estimate_capacity.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o seqreader.o seqreader.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o mmscanner.o mmscanner.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o omp_hack.o omp_hack.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o utilities.o utilities.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o estimate_capacity estimate_capacity.o seqreader.o mmscanner.o omp_hack.o utilities.o\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o build_db.o build_db.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o mmap_file.o mmap_file.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o compact_hash.o compact_hash.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o taxonomy.o taxonomy.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o build_db build_db.o mmap_file.o compact_hash.o taxonomy.o seqreader.o mmscanner.o omp_hack.o utilities.o\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o classify.o classify.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o reports.o reports.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o hyperloglogplus.o hyperloglogplus.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o aa_translate.o aa_translate.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o classify classify.o reports.o hyperloglogplus.o mmap_file.o compact_hash.o taxonomy.o seqreader.o mmscanner.o omp_hack.o aa_translate.o utilities.o\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o dump_table.o dump_table.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o dump_table dump_table.o mmap_file.o compact_hash.o omp_hack.o taxonomy.o reports.o hyperloglogplus.o\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o lookup_accession_numbers.o lookup_accession_numbers.cc\n",
      "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o lookup_accession_numbers lookup_accession_numbers.o mmap_file.o omp_hack.o utilities.o\n",
      "cp estimate_capacity build_db classify dump_table lookup_accession_numbers /content/kraken2/KRAKEN/\n",
      "make: Leaving directory '/content/kraken2/src'\n",
      "\n",
      "Kraken 2 installation complete.\n",
      "\n",
      "To make things easier for you, you may want to copy/symlink the following\n",
      "files into a directory in your PATH:\n",
      "  /content/kraken2/KRAKEN/kraken2\n",
      "  /content/kraken2/KRAKEN/kraken2-build\n",
      "  /content/kraken2/KRAKEN/kraken2-inspect\n"
     ]
    }
   ],
   "source": [
    "%cd kraken2/\n",
    "! ./install_kraken2.sh KRAKEN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "kzdAO1m0hBa6",
    "outputId": "4eb204ff-a413-457c-aa08-0cf54a73c255"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/content\n"
     ]
    }
   ],
   "source": [
    "%cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "3hEN73eKhKUl",
    "outputId": "637c5eb9-1a6b-4d0c-8af9-673370bff868"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "KRAKEN\tkraken2  sample_data\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "gR7Y2_xtfKuN",
    "outputId": "eab42a66-ddd0-408c-e3ac-60c5f9456f00"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Masking low-complexity regions of new file...Unable to find dustmasker in path, can't mask low-complexity sequences\n",
      "Masking low-complexity regions of new file...Unable to find dustmasker in path, can't mask low-complexity sequences\n",
      "Masking low-complexity regions of new file...Unable to find dustmasker in path, can't mask low-complexity sequences\n",
      "Creating sequence ID to taxonomy ID map (step 1)...\n",
      "Sequence ID to taxonomy ID map already present, skipping map creation.\n",
      "Estimating required capacity (step 2)...\n",
      "Estimated hash table requirement: 514023128 bytes\n",
      "Capacity estimation complete. [23.191s]\n",
      "Building database files (step 3)...\n",
      "Hash table already present, skipping database file build.\n",
      "Database construction complete. [Total: 23.213s]\n"
     ]
    }
   ],
   "source": [
    "!kraken2/KRAKEN/kraken2-build --download-library viral --db ViralAttempt3\n",
    "!kraken2/KRAKEN/kraken2-build --db ViralAttempt3 --download-taxonomy --use-ftp\n",
    "\n",
    "!wget ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/viral.*.1.genomic.fna.gz\n",
    "!gunzip viral.1.1.genomic.fna.gz\n",
    "!gunzip viral.2.1.genomic.fna.gz\n",
    "!gunzip viral.3.1.genomic.fna.gz\n",
    "\n",
    "!kraken2/KRAKEN/kraken2-build --add-to-library viral.1.1.genomic.fna -db ViralAttempt3\n",
    "!kraken2/KRAKEN/kraken2-build --add-to-library viral.2.1.genomic.fna -db ViralAttempt3\n",
    "!kraken2/KRAKEN/kraken2-build --add-to-library viral.3.1.genomic.fna -db ViralAttempt3\n",
    "!kraken2/KRAKEN/kraken2-build --build --db ViralAttempt3 --threads 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "_6WZOo7DlNkY",
    "outputId": "d63fec80-3c0a-42af-ac21-0f8551b8e678"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gzip: 14.1.fastq.gz: No such file or directory\n",
      "gzip: 14.2.fastq.gz: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!gunzip 14.1.fastq.gz\n",
    "!gunzip 14.2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "fUAjQUNchTUV",
    "outputId": "f0d44af3-e905-41b6-9aff-8f04e8b62872"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information... done.\n",
      "10000168 sequences (2000.03 Mbp) processed in 140.293s (4276.8 Kseq/m, 855.37 Mbp/m).\n",
      "  22 sequences classified (0.00%)\n",
      "  10000146 sequences unclassified (100.00%)\n"
     ]
    }
   ],
   "source": [
    "!kraken2/KRAKEN/kraken2 --threads 1 --db ViralAttempt3 --paired --classified-out cseqs1#.fq 14.1.fastq  14.2.fastq  --confidence 1 --report REPORT_NCBI_VIRAL1.report  -output steam.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "VB6DFTNgiG98",
    "outputId": "6e7999c3-c784-40a2-841e-f7580de06b2c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-03-27 13:34:03--  https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh\n",
      "Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8303, ...\n",
      "Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 85055499 (81M) [application/x-sh]\n",
      "Saving to: ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’\n",
      "\n",
      "Miniconda3-py37_4.8 100%[===================>]  81.12M   142MB/s    in 0.6s    \n",
      "\n",
      "2021-03-27 13:34:04 (142 MB/s) - ‘Miniconda3-py37_4.8.2-Linux-x86_64.sh’ saved [85055499/85055499]\n",
      "\n",
      "--2021-03-27 13:34:04--  http://!/\n",
      "Resolving ! (!)... failed: Name or service not known.\n",
      "wget: unable to resolve host address ‘!’\n",
      "--2021-03-27 13:34:04--  http://chmod/\n",
      "Resolving chmod (chmod)... failed: Name or service not known.\n",
      "wget: unable to resolve host address ‘chmod’\n",
      "--2021-03-27 13:34:04--  http://+x/\n",
      "Resolving +x (+x)... failed: Name or service not known.\n",
      "wget: unable to resolve host address ‘+x’\n",
      "--2021-03-27 13:34:04--  http://miniconda3-py37_4.8.2-linux-x86_64.sh/\n",
      "Resolving miniconda3-py37_4.8.2-linux-x86_64.sh (miniconda3-py37_4.8.2-linux-x86_64.sh)... failed: Name or service not known.\n",
      "wget: unable to resolve host address ‘miniconda3-py37_4.8.2-linux-x86_64.sh’\n",
      "FINISHED --2021-03-27 13:34:04--\n",
      "Total wall clock time: 1.0s\n",
      "Downloaded: 1 files, 81M in 0.6s (142 MB/s)\n",
      "PREFIX=/usr/local\n",
      "Unpacking payload ...\n",
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n",
      "Solving environment: - \b\b\\ \b\bdone\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /usr/local\n",
      "\n",
      "  added / updated specs:\n",
      "    - _libgcc_mutex==0.1=main\n",
      "    - asn1crypto==1.3.0=py37_0\n",
      "    - ca-certificates==2020.1.1=0\n",
      "    - certifi==2019.11.28=py37_0\n",
      "    - cffi==1.14.0=py37h2e261b9_0\n",
      "    - chardet==3.0.4=py37_1003\n",
      "    - conda-package-handling==1.6.0=py37h7b6447c_0\n",
      "    - conda==4.8.2=py37_0\n",
      "    - cryptography==2.8=py37h1ba5d50_0\n",
      "    - idna==2.8=py37_0\n",
      "    - ld_impl_linux-64==2.33.1=h53a641e_7\n",
      "    - libedit==3.1.20181209=hc058e9b_0\n",
      "    - libffi==3.2.1=hd88cf55_4\n",
      "    - libgcc-ng==9.1.0=hdf63c60_0\n",
      "    - libstdcxx-ng==9.1.0=hdf63c60_0\n",
      "    - ncurses==6.2=he6710b0_0\n",
      "    - openssl==1.1.1d=h7b6447c_4\n",
      "    - pip==20.0.2=py37_1\n",
      "    - pycosat==0.6.3=py37h7b6447c_0\n",
      "    - pycparser==2.19=py37_0\n",
      "    - pyopenssl==19.1.0=py37_0\n",
      "    - pysocks==1.7.1=py37_0\n",
      "    - python==3.7.6=h0371630_2\n",
      "    - readline==7.0=h7b6447c_5\n",
      "    - requests==2.22.0=py37_1\n",
      "    - ruamel_yaml==0.15.87=py37h7b6447c_0\n",
      "    - setuptools==45.2.0=py37_0\n",
      "    - six==1.14.0=py37_0\n",
      "    - sqlite==3.31.1=h7b6447c_0\n",
      "    - tk==8.6.8=hbc83047_0\n",
      "    - tqdm==4.42.1=py_0\n",
      "    - urllib3==1.25.8=py37_0\n",
      "    - wheel==0.34.2=py37_0\n",
      "    - xz==5.2.4=h14c3975_4\n",
      "    - yaml==0.1.7=had09818_2\n",
      "    - zlib==1.2.11=h7b6447c_3\n",
      "\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n",
      "  asn1crypto         pkgs/main/linux-64::asn1crypto-1.3.0-py37_0\n",
      "  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.1.1-0\n",
      "  certifi            pkgs/main/linux-64::certifi-2019.11.28-py37_0\n",
      "  cffi               pkgs/main/linux-64::cffi-1.14.0-py37h2e261b9_0\n",
      "  chardet            pkgs/main/linux-64::chardet-3.0.4-py37_1003\n",
      "  conda              pkgs/main/linux-64::conda-4.8.2-py37_0\n",
      "  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.6.0-py37h7b6447c_0\n",
      "  cryptography       pkgs/main/linux-64::cryptography-2.8-py37h1ba5d50_0\n",
      "  idna               pkgs/main/linux-64::idna-2.8-py37_0\n",
      "  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7\n",
      "  libedit            pkgs/main/linux-64::libedit-3.1.20181209-hc058e9b_0\n",
      "  libffi             pkgs/main/linux-64::libffi-3.2.1-hd88cf55_4\n",
      "  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0\n",
      "  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0\n",
      "  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_0\n",
      "  openssl            pkgs/main/linux-64::openssl-1.1.1d-h7b6447c_4\n",
      "  pip                pkgs/main/linux-64::pip-20.0.2-py37_1\n",
      "  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py37h7b6447c_0\n",
      "  pycparser          pkgs/main/linux-64::pycparser-2.19-py37_0\n",
      "  pyopenssl          pkgs/main/linux-64::pyopenssl-19.1.0-py37_0\n",
      "  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py37_0\n",
      "  python             pkgs/main/linux-64::python-3.7.6-h0371630_2\n",
      "  readline           pkgs/main/linux-64::readline-7.0-h7b6447c_5\n",
      "  requests           pkgs/main/linux-64::requests-2.22.0-py37_1\n",
      "  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py37h7b6447c_0\n",
      "  setuptools         pkgs/main/linux-64::setuptools-45.2.0-py37_0\n",
      "  six                pkgs/main/linux-64::six-1.14.0-py37_0\n",
      "  sqlite             pkgs/main/linux-64::sqlite-3.31.1-h7b6447c_0\n",
      "  tk                 pkgs/main/linux-64::tk-8.6.8-hbc83047_0\n",
      "  tqdm               pkgs/main/noarch::tqdm-4.42.1-py_0\n",
      "  urllib3            pkgs/main/linux-64::urllib3-1.25.8-py37_0\n",
      "  wheel              pkgs/main/linux-64::wheel-0.34.2-py37_0\n",
      "  xz                 pkgs/main/linux-64::xz-5.2.4-h14c3975_4\n",
      "  yaml               pkgs/main/linux-64::yaml-0.1.7-had09818_2\n",
      "  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3\n",
      "\n",
      "\n",
      "Preparing transaction: / \b\b- \b\b\\ \b\b| \b\bdone\n",
      "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "installation finished.\n",
      "WARNING:\n",
      "    You currently have a PYTHONPATH environment variable set. This may cause\n",
      "    unexpected behavior when running the Python interpreter in Miniconda3.\n",
      "    For best results, please verify that your PYTHONPATH only points to\n",
      "    directories of packages that are compatible with the Python interpreter\n",
      "    in Miniconda3: /usr/local\n"
     ]
    }
   ],
   "source": [
    "!wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.8.2-Linux-x86_64.sh ! chmod +x Miniconda3-py37_4.8.2-Linux-x86_64.sh\n",
    "!bash ./Miniconda3-py37_4.8.2-Linux-x86_64.sh -b -f -p /usr/local\n",
    "\n",
    "import sys \n",
    "\n",
    "sys.path.append('/usr/local/lib/python3.7/site-packages/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "eZc4c5kZmyrO",
    "outputId": "796e67f1-b454-4ad0-8d4b-984b5ca165da"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "Solving environment: / \b\b- \b\b\\ \b\b| \b\bdone\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Requirement already satisfied: deepTools==2.5.3 in /usr/local/lib/python3.7/site-packages (2.5.3)\n",
      "Requirement already satisfied: py2bit>=0.2.0 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (0.3.0)\n",
      "Requirement already satisfied: pyBigWig>=0.2.1 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (0.3.18)\n",
      "Requirement already satisfied: pysam>=0.8.2 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (0.16.0.1)\n",
      "Requirement already satisfied: numpy>=1.9.0 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (1.20.1)\n",
      "Requirement already satisfied: numpydoc>=0.5 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (1.1.0)\n",
      "Requirement already satisfied: scipy>=0.17.0 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (1.6.2)\n",
      "Requirement already satisfied: matplotlib>=1.4.0 in /usr/local/lib/python3.7/site-packages (from deepTools==2.5.3) (3.4.0)\n",
      "Requirement already satisfied: sphinx>=1.6.5 in /usr/local/lib/python3.7/site-packages (from numpydoc>=0.5->deepTools==2.5.3) (3.5.3)\n",
      "Requirement already satisfied: Jinja2>=2.3 in /usr/local/lib/python3.7/site-packages (from numpydoc>=0.5->deepTools==2.5.3) (2.11.3)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.7/site-packages (from matplotlib>=1.4.0->deepTools==2.5.3) (8.1.2)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib>=1.4.0->deepTools==2.5.3) (2.4.7)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/site-packages (from matplotlib>=1.4.0->deepTools==2.5.3) (1.3.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.7/site-packages (from matplotlib>=1.4.0->deepTools==2.5.3) (2.8.1)\n",
      "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/site-packages (from matplotlib>=1.4.0->deepTools==2.5.3) (0.10.0)\n",
      "Requirement already satisfied: snowballstemmer>=1.1 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2.1.0)\n",
      "Requirement already satisfied: Pygments>=2.0 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2.8.1)\n",
      "Requirement already satisfied: docutils>=0.12 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (0.16)\n",
      "Requirement already satisfied: imagesize in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.2.0)\n",
      "Requirement already satisfied: sphinxcontrib-htmlhelp in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.0.3)\n",
      "Requirement already satisfied: alabaster<0.8,>=0.7 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (0.7.12)\n",
      "Requirement already satisfied: sphinxcontrib-applehelp in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.0.2)\n",
      "Requirement already satisfied: packaging in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (20.9)\n",
      "Requirement already satisfied: sphinxcontrib-devhelp in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.0.2)\n",
      "Requirement already satisfied: sphinxcontrib-serializinghtml in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.1.4)\n",
      "Requirement already satisfied: sphinxcontrib-qthelp in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.0.3)\n",
      "Requirement already satisfied: babel>=1.3 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2.9.0)\n",
      "Requirement already satisfied: requests>=2.5.0 in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2.22.0)\n",
      "Requirement already satisfied: setuptools in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (45.2.0.post20200210)\n",
      "Requirement already satisfied: sphinxcontrib-jsmath in /usr/local/lib/python3.7/site-packages (from sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.0.1)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/site-packages (from Jinja2>=2.3->numpydoc>=0.5->deepTools==2.5.3) (1.1.1)\n",
      "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.7->matplotlib>=1.4.0->deepTools==2.5.3) (1.14.0)\n",
      "Requirement already satisfied: pytz>=2015.7 in /usr/local/lib/python3.7/site-packages (from babel>=1.3->sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2021.1)\n",
      "Requirement already satisfied: idna<2.9,>=2.5 in /usr/local/lib/python3.7/site-packages (from requests>=2.5.0->sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2.8)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/site-packages (from requests>=2.5.0->sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (1.25.8)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/site-packages (from requests>=2.5.0->sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (2020.12.5)\n",
      "Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/local/lib/python3.7/site-packages (from requests>=2.5.0->sphinx>=1.6.5->numpydoc>=0.5->deepTools==2.5.3) (3.0.4)\n",
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
      "Solving environment: | \b\b/ \b\b- \b\b\\ \b\bdone\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "Requirement already satisfied: rseqc in /usr/local/lib/python3.7/site-packages (4.0.0)\n",
      "Requirement already satisfied: bx-python in /usr/local/lib/python3.7/site-packages (from rseqc) (0.8.11)\n",
      "Requirement already satisfied: pyBigWig in /usr/local/lib/python3.7/site-packages (from rseqc) (0.3.18)\n",
      "Requirement already satisfied: cython>=0.17 in /usr/local/lib/python3.7/site-packages (from rseqc) (0.29.22)\n",
      "Requirement already satisfied: pysam in /usr/local/lib/python3.7/site-packages (from rseqc) (0.16.0.1)\n",
      "Requirement already satisfied: numpy in /usr/local/lib/python3.7/site-packages (from rseqc) (1.20.1)\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda kallisto -y\n",
    "!conda install -c bioconda hisat2 -y\n",
    "\n",
    "!conda install -c bioconda samtools -y\n",
    "\n",
    "!pip install deepTools==2.5.3\n",
    "\n",
    "!conda install -c bioconda subread -y\n",
    "\n",
    "\n",
    "\n",
    "!conda install -c bioconda sra-tools -y\n",
    "!pip install rseqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "K_ceFzPcGiCB",
    "outputId": "7c925f8b-0fde-4fd8-fcd1-d7f76f62fab9"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-03-27 14:39:49--  ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh38.100.gtf.gz\n",
      "           => ‘Homo_sapiens.GRCh38.100.gtf.gz’\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.139\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.139|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-100/gtf/homo_sapiens ... done.\n",
      "==> SIZE Homo_sapiens.GRCh38.100.gtf.gz ... 46973686\n",
      "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.100.gtf.gz ... done.\n",
      "Length: 46973686 (45M) (unauthoritative)\n",
      "\n",
      "Homo_sapiens.GRCh38 100%[===================>]  44.80M  15.3MB/s    in 2.9s    \n",
      "\n",
      "2021-03-27 14:39:54 (15.3 MB/s) - ‘Homo_sapiens.GRCh38.100.gtf.gz’ saved [46973686]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh38.100.gtf.gz\n",
    "!gunzip Homo_sapiens.GRCh38.100.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "58koP2WvId4o",
    "outputId": "9bff3fb4-9fcc-43b6-ab3d-c54b94261051"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-03-27 14:42:40--  ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
      "           => ‘Homo_sapiens.GRCh38.cdna.all.fa.gz’\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.197.76\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.197.76|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-100/fasta/homo_sapiens/cdna ... done.\n",
      "==> SIZE Homo_sapiens.GRCh38.cdna.all.fa.gz ... 69918383\n",
      "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.cdna.all.fa.gz ... done.\n",
      "Length: 69918383 (67M) (unauthoritative)\n",
      "\n",
      "Homo_sapiens.GRCh38 100%[===================>]  66.68M  17.4MB/s    in 4.0s    \n",
      "\n",
      "2021-03-27 14:42:46 (16.7 MB/s) - ‘Homo_sapiens.GRCh38.cdna.all.fa.gz’ saved [69918383]\n",
      "\n",
      "Settings:\n",
      "  Output files: \"Homo_sapiens_assembly.*.ht2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Local offset rate: 3 (one in 8)\n",
      "  Local fTable chars: 6\n",
      "  Local sequence length: 57344\n",
      "  Local sequence overlap between two consecutive indexes: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  Homo_sapiens.GRCh38.cdna.all.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:03\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:02\n",
      "  Time to read SNPs and splice sites: 00:00:00\n",
      "Using parameters --bmax 15089985 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 15089985 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:20\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:04\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:04\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 44 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 44 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 44 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 6, merged 22; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 2, merged 1; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.11007e+07 (target: 15089984)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering GFM loop\n",
      "Getting block 1 of 29\n",
      "  Reserving size (15089985) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "Getting block 2 of 29\n",
      "  Reserving size (15089985) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "Getting block 3 of 29\n",
      "  Reserving size (15089985) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "Getting block 4 of 29\n",
      "  Reserving size (15089985) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 1: 10%\n",
      "  bucket 2: 10%\n",
      "  bucket 3: 10%\n",
      "  bucket 4: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 2: 20%\n",
      "  bucket 3: 20%\n",
      "  bucket 4: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 2: 30%\n",
      "  bucket 3: 30%\n",
      "  bucket 4: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 2: 40%\n",
      "  bucket 3: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 4: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 3: 50%\n",
      "  bucket 4: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 3: 60%\n",
      "  bucket 4: 60%\n",
      "  bucket 1: 80%\n",
      "  bucket 2: 70%\n",
      "  bucket 3: 70%\n",
      "  bucket 4: 70%\n",
      "  bucket 1: 90%\n",
      "  bucket 2: 80%\n",
      "  bucket 3: 80%\n",
      "  bucket 4: 80%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 12924294 for bucket 1\n",
      "  (Using difference cover)\n",
      "  bucket 2: 90%\n",
      "  bucket 3: 90%\n",
      "  bucket 4: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 14448884 for bucket 2\n",
      "  (Using difference cover)\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 14337530 for bucket 3\n",
      "  (Using difference cover)\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 8137383 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:15\n",
      "Returning block of 8137384 for bucket 4\n",
      "Getting block 5 of 29\n",
      "  Reserving size (15089985) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 14857068 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:49\n",
      "Returning block of 12924295 for bucket 1\n",
      "Getting block 6 of 29\n",
      "  Reserving size (15089985) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  Sorting block time: 00:02:06\n",
      "Returning block of 14448885 for bucket 2\n",
      "  bucket 6: 90%\n",
      "Getting block 7 of 29\n",
      "  Reserving size (15089985) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 14985754 for bucket 6\n",
      "  (Using difference cover)\n",
      "  bucket 7: 10%\n",
      "  Sorting block time: 00:02:10\n",
      "Returning block of 14337531 for bucket 3\n",
      "  bucket 7: 20%\n",
      "Getting block 8 of 29\n",
      "  Reserving size (15089985) for bucket 8\n",
      "  Calculating Z arrays for bucket 8\n",
      "  Entering block accumulator loop for bucket 8:\n",
      "  bucket 7: 30%\n",
      "  bucket 8: 10%\n",
      "  bucket 7: 40%\n",
      "  bucket 8: 20%\n",
      "  bucket 8: 30%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 8: 40%\n",
      "  bucket 8: 50%\n",
      "  bucket 7: 70%\n",
      "  bucket 8: 60%\n",
      "  bucket 7: 80%\n",
      "  bucket 8: 70%\n",
      "  bucket 7: 90%\n",
      "  bucket 8: 80%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 11799596 for bucket 7\n",
      "  (Using difference cover)\n",
      "  bucket 8: 90%\n",
      "  bucket 8: 100%\n",
      "  Sorting block of length 8815748 for bucket 8\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:18\n",
      "Returning block of 14857069 for bucket 5\n",
      "Getting block 9 of 29\n",
      "  Reserving size (15089985) for bucket 9\n",
      "  Calculating Z arrays for bucket 9\n",
      "  Entering block accumulator loop for bucket 9:\n",
      "  Sorting block time: 00:01:21\n",
      "Returning block of 8815749 for bucket 8\n",
      "  bucket 9: 10%\n",
      "Getting block 10 of 29\n",
      "  Reserving size (15089985) for bucket 10\n",
      "  Calculating Z arrays for bucket 10\n",
      "  Entering block accumulator loop for bucket 10:\n",
      "  bucket 9: 20%\n",
      "  bucket 10: 10%\n",
      "  bucket 9: 30%\n",
      "  bucket 10: 20%\n",
      "  bucket 9: 40%\n",
      "  bucket 10: 30%\n",
      "  bucket 9: 50%\n",
      "  bucket 10: 40%\n",
      "  bucket 9: 60%\n",
      "  bucket 10: 50%\n",
      "  bucket 9: 70%\n",
      "  bucket 10: 60%\n",
      "  bucket 9: 80%\n",
      "  bucket 10: 70%\n",
      "  bucket 9: 90%\n",
      "  bucket 10: 80%\n",
      "  bucket 9: 100%\n",
      "  Sorting block of length 6775355 for bucket 9\n",
      "  (Using difference cover)\n",
      "  bucket 10: 90%\n",
      "  Sorting block time: 00:01:44\n",
      "Returning block of 11799597 for bucket 7\n",
      "  bucket 10: 100%\n",
      "  Sorting block of length 12332396 for bucket 10\n",
      "  (Using difference cover)\n",
      "Getting block 11 of 29\n",
      "  Reserving size (15089985) for bucket 11\n",
      "  Calculating Z arrays for bucket 11\n",
      "  Entering block accumulator loop for bucket 11:\n",
      "  bucket 11: 10%\n",
      "  bucket 11: 20%\n",
      "  Sorting block time: 00:02:12\n",
      "Returning block of 14985755 for bucket 6\n",
      "  bucket 11: 30%\n",
      "Getting block 12 of 29\n",
      "  Reserving size (15089985) for bucket 12\n",
      "  Calculating Z arrays for bucket 12\n",
      "  Entering block accumulator loop for bucket 12:\n",
      "  bucket 11: 40%\n",
      "  bucket 12: 10%\n",
      "  bucket 11: 50%\n",
      "  bucket 12: 20%\n",
      "  bucket 11: 60%\n",
      "  bucket 12: 30%\n",
      "  bucket 11: 70%\n",
      "  bucket 12: 40%\n",
      "  bucket 11: 80%\n",
      "  bucket 12: 50%\n",
      "  bucket 11: 90%\n",
      "  bucket 12: 60%\n",
      "  bucket 11: 100%\n",
      "  Sorting block of length 11405394 for bucket 11\n",
      "  (Using difference cover)\n",
      "  bucket 12: 70%\n",
      "  bucket 12: 80%\n",
      "  bucket 12: 90%\n",
      "  bucket 12: 100%\n",
      "  Sorting block of length 6740553 for bucket 12\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:04\n",
      "Returning block of 6775356 for bucket 9\n",
      "Getting block 13 of 29\n",
      "  Reserving size (15089985) for bucket 13\n",
      "  Calculating Z arrays for bucket 13\n",
      "  Entering block accumulator loop for bucket 13:\n",
      "  bucket 13: 10%\n",
      "  bucket 13: 20%\n",
      "  bucket 13: 30%\n",
      "  bucket 13: 40%\n",
      "  bucket 13: 50%\n",
      "  bucket 13: 60%\n",
      "  bucket 13: 70%\n",
      "  bucket 13: 80%\n",
      "  bucket 13: 90%\n",
      "  bucket 13: 100%\n",
      "  Sorting block of length 8741476 for bucket 13\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:03\n",
      "Returning block of 6740554 for bucket 12\n",
      "Getting block 14 of 29\n",
      "  Reserving size (15089985) for bucket 14\n",
      "  Calculating Z arrays for bucket 14\n",
      "  Entering block accumulator loop for bucket 14:\n",
      "  bucket 14: 10%\n",
      "  bucket 14: 20%\n",
      "  bucket 14: 30%\n",
      "  bucket 14: 40%\n",
      "  bucket 14: 50%\n",
      "  bucket 14: 60%\n",
      "  bucket 14: 70%\n",
      "  bucket 14: 80%\n",
      "  bucket 14: 90%\n",
      "  bucket 14: 100%\n",
      "  Sorting block of length 13139612 for bucket 14\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:55\n",
      "Returning block of 12332397 for bucket 10\n",
      "Getting block 15 of 29\n",
      "  Reserving size (15089985) for bucket 15\n",
      "  Calculating Z arrays for bucket 15\n",
      "  Entering block accumulator loop for bucket 15:\n",
      "  bucket 15: 10%\n",
      "  bucket 15: 20%\n",
      "  bucket 15: 30%\n",
      "  bucket 15: 40%\n",
      "  Sorting block time: 00:01:47\n",
      "Returning block of 11405395 for bucket 11\n",
      "Getting block 16 of 29\n",
      "  Reserving size (15089985) for bucket 16\n",
      "  Calculating Z arrays for bucket 16\n",
      "  Entering block accumulator loop for bucket 16:\n",
      "  bucket 15: 50%\n",
      "  bucket 16: 10%\n",
      "  bucket 15: 60%\n",
      "  bucket 16: 20%\n",
      "  bucket 15: 70%\n",
      "  bucket 16: 30%\n",
      "  bucket 15: 80%\n",
      "  bucket 16: 40%\n",
      "  bucket 15: 90%\n",
      "  bucket 16: 50%\n",
      "  bucket 15: 100%\n",
      "  Sorting block of length 11461268 for bucket 15\n",
      "  (Using difference cover)\n",
      "  bucket 16: 60%\n",
      "  bucket 16: 70%\n",
      "  bucket 16: 80%\n",
      "  bucket 16: 90%\n",
      "  bucket 16: 100%\n",
      "  Sorting block of length 14583721 for bucket 16\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:21\n",
      "Returning block of 8741477 for bucket 13\n",
      "Getting block 17 of 29\n",
      "  Reserving size (15089985) for bucket 17\n",
      "  Calculating Z arrays for bucket 17\n",
      "  Entering block accumulator loop for bucket 17:\n",
      "  bucket 17: 10%\n",
      "  bucket 17: 20%\n",
      "  bucket 17: 30%\n",
      "  bucket 17: 40%\n",
      "  bucket 17: 50%\n",
      "  bucket 17: 60%\n",
      "  bucket 17: 70%\n",
      "  bucket 17: 80%\n",
      "  bucket 17: 90%\n",
      "  bucket 17: 100%\n",
      "  Sorting block of length 11148647 for bucket 17\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:01\n",
      "Returning block of 13139613 for bucket 14\n",
      "Getting block 18 of 29\n",
      "  Reserving size (15089985) for bucket 18\n",
      "  Calculating Z arrays for bucket 18\n",
      "  Entering block accumulator loop for bucket 18:\n",
      "  bucket 18: 10%\n",
      "  bucket 18: 20%\n",
      "  bucket 18: 30%\n",
      "  bucket 18: 40%\n",
      "  Sorting block time: 00:01:44\n",
      "Returning block of 11461269 for bucket 15\n",
      "  bucket 18: 50%\n",
      "Getting block 19 of 29\n",
      "  Reserving size (15089985) for bucket 19\n",
      "  Calculating Z arrays for bucket 19\n",
      "  Entering block accumulator loop for bucket 19:\n",
      "  bucket 18: 60%\n",
      "  bucket 19: 10%\n",
      "  bucket 18: 70%\n",
      "  bucket 19: 20%\n",
      "  bucket 18: 80%\n",
      "  bucket 19: 30%\n",
      "  bucket 18: 90%\n",
      "  bucket 19: 40%\n",
      "  bucket 18: 100%\n",
      "  Sorting block of length 14129981 for bucket 18\n",
      "  (Using difference cover)\n",
      "  bucket 19: 50%\n",
      "  bucket 19: 60%\n",
      "  bucket 19: 70%\n",
      "  bucket 19: 80%\n",
      "  bucket 19: 90%\n",
      "  bucket 19: 100%\n",
      "  Sorting block of length 5161522 for bucket 19\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:15\n",
      "Returning block of 14583722 for bucket 16\n",
      "  Sorting block time: 00:01:43\n",
      "Returning block of 11148648 for bucket 17\n",
      "Getting block 20 of 29\n",
      "  Reserving size (15089985) for bucket 20\n",
      "  Calculating Z arrays for bucket 20\n",
      "  Entering block accumulator loop for bucket 20:\n",
      "Getting block 21 of 29\n",
      "  Reserving size (15089985) for bucket 21\n",
      "  Calculating Z arrays for bucket 21\n",
      "  Entering block accumulator loop for bucket 21:\n",
      "  bucket 20: 10%\n",
      "  bucket 21: 10%\n",
      "  bucket 20: 20%\n",
      "  bucket 21: 20%\n",
      "  bucket 20: 30%\n",
      "  bucket 21: 30%\n",
      "  bucket 20: 40%\n",
      "  bucket 21: 40%\n",
      "  bucket 20: 50%\n",
      "  bucket 21: 50%\n",
      "  bucket 20: 60%\n",
      "  bucket 21: 60%\n",
      "  bucket 20: 70%\n",
      "  bucket 21: 70%\n",
      "  bucket 20: 80%\n",
      "  bucket 21: 80%\n",
      "  bucket 20: 90%\n",
      "  bucket 21: 90%\n",
      "  bucket 20: 100%\n",
      "  Sorting block of length 14704877 for bucket 20\n",
      "  (Using difference cover)\n",
      "  bucket 21: 100%\n",
      "  Sorting block of length 4474867 for bucket 21\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:50\n",
      "Returning block of 5161523 for bucket 19\n",
      "Getting block 22 of 29\n",
      "  Reserving size (15089985) for bucket 22\n",
      "  Calculating Z arrays for bucket 22\n",
      "  Entering block accumulator loop for bucket 22:\n",
      "  bucket 22: 10%\n",
      "  bucket 22: 20%\n",
      "  bucket 22: 30%\n",
      "  bucket 22: 40%\n",
      "  bucket 22: 50%\n",
      "  bucket 22: 60%\n",
      "  bucket 22: 70%\n",
      "  bucket 22: 80%\n",
      "  bucket 22: 90%\n",
      "  bucket 22: 100%\n",
      "  Sorting block of length 14136373 for bucket 22\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:40\n",
      "Returning block of 4474868 for bucket 21\n",
      "Getting block 23 of 29\n",
      "  Reserving size (15089985) for bucket 23\n",
      "  Calculating Z arrays for bucket 23\n",
      "  Entering block accumulator loop for bucket 23:\n",
      "  bucket 23: 10%\n",
      "  bucket 23: 20%\n",
      "  bucket 23: 30%\n",
      "  bucket 23: 40%\n",
      "  bucket 23: 50%\n",
      "  bucket 23: 60%\n",
      "  bucket 23: 70%\n",
      "  bucket 23: 80%\n",
      "  bucket 23: 90%\n",
      "  bucket 23: 100%\n",
      "  Sorting block of length 13266825 for bucket 23\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:10\n",
      "Returning block of 14129982 for bucket 18\n",
      "Getting block 24 of 29\n",
      "  Reserving size (15089985) for bucket 24\n",
      "  Calculating Z arrays for bucket 24\n",
      "  Entering block accumulator loop for bucket 24:\n",
      "  bucket 24: 10%\n",
      "  bucket 24: 20%\n",
      "  bucket 24: 30%\n",
      "  bucket 24: 40%\n",
      "  bucket 24: 50%\n",
      "  bucket 24: 60%\n",
      "  bucket 24: 70%\n",
      "  bucket 24: 80%\n",
      "  bucket 24: 90%\n",
      "  bucket 24: 100%\n",
      "  Sorting block of length 14447157 for bucket 24\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:08\n",
      "Returning block of 14704878 for bucket 20\n",
      "Getting block 25 of 29\n",
      "  Reserving size (15089985) for bucket 25\n",
      "  Calculating Z arrays for bucket 25\n",
      "  Entering block accumulator loop for bucket 25:\n",
      "  bucket 25: 10%\n",
      "  bucket 25: 20%\n",
      "  bucket 25: 30%\n",
      "  bucket 25: 40%\n",
      "  bucket 25: 50%\n",
      "  bucket 25: 60%\n",
      "  bucket 25: 70%\n",
      "  bucket 25: 80%\n",
      "  Sorting block time: 00:02:02\n",
      "Returning block of 14136374 for bucket 22\n",
      "  bucket 25: 90%\n",
      "Getting block 26 of 29\n",
      "  Reserving size (15089985) for bucket 26\n",
      "  Calculating Z arrays for bucket 26\n",
      "  Entering block accumulator loop for bucket 26:\n",
      "  bucket 25: 100%\n",
      "  Sorting block of length 10358202 for bucket 25\n",
      "  (Using difference cover)\n",
      "  bucket 26: 10%\n",
      "  bucket 26: 20%\n",
      "  bucket 26: 30%\n",
      "  bucket 26: 40%\n",
      "  bucket 26: 50%\n",
      "  bucket 26: 60%\n",
      "  bucket 26: 70%\n",
      "  bucket 26: 80%\n",
      "  bucket 26: 90%\n",
      "  bucket 26: 100%\n",
      "  Sorting block of length 9099205 for bucket 26\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:57\n",
      "Returning block of 13266826 for bucket 23\n",
      "Getting block 27 of 29\n",
      "  Reserving size (15089985) for bucket 27\n",
      "  Calculating Z arrays for bucket 27\n",
      "  Entering block accumulator loop for bucket 27:\n",
      "  bucket 27: 10%\n",
      "  bucket 27: 20%\n",
      "  bucket 27: 30%\n",
      "  bucket 27: 40%\n",
      "  bucket 27: 50%\n",
      "  bucket 27: 60%\n",
      "  bucket 27: 70%\n",
      "  bucket 27: 80%\n",
      "  bucket 27: 90%\n",
      "  bucket 27: 100%\n",
      "  Sorting block of length 7595047 for bucket 27\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:02:11\n",
      "Returning block of 14447158 for bucket 24\n",
      "Getting block 28 of 29\n",
      "  Reserving size (15089985) for bucket 28\n",
      "  Calculating Z arrays for bucket 28\n",
      "  Entering block accumulator loop for bucket 28:\n",
      "  bucket 28: 10%\n",
      "  bucket 28: 20%\n",
      "  bucket 28: 30%\n",
      "  bucket 28: 40%\n",
      "  bucket 28: 50%\n",
      "  bucket 28: 60%\n",
      "  Sorting block time: 00:01:35\n",
      "Returning block of 10358203 for bucket 25\n",
      "  bucket 28: 70%\n",
      "Getting block 29 of 29\n",
      "  Reserving size (15089985) for bucket 29\n",
      "  Calculating Z arrays for bucket 29\n",
      "  Entering block accumulator loop for bucket 29:\n",
      "  bucket 29: 10%\n",
      "  bucket 28: 80%\n",
      "  bucket 29: 20%\n",
      "  bucket 29: 30%\n",
      "  bucket 28: 90%\n",
      "  bucket 29: 40%\n",
      "  bucket 28: 100%\n",
      "  Sorting block of length 12429391 for bucket 28\n",
      "  (Using difference cover)\n",
      "  bucket 29: 50%\n",
      "  Sorting block time: 00:01:23\n",
      "Returning block of 9099206 for bucket 26\n",
      "  bucket 29: 60%\n",
      "  bucket 29: 70%\n",
      "  bucket 29: 80%\n",
      "  bucket 29: 90%\n",
      "  bucket 29: 100%\n",
      "  Sorting block of length 5481522 for bucket 29\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:01:02\n",
      "Returning block of 7595048 for bucket 27\n",
      "  Sorting block time: 00:00:22\n",
      "Returning block of 5481523 for bucket 29\n",
      "  Sorting block time: 00:00:48\n",
      "Returning block of 12429392 for bucket 28\n",
      "Exited GFM loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 83422230\n",
      "fchr[G]: 162718329\n",
      "fchr[T]: 243227109\n",
      "fchr[$]: 321919676\n",
      "Exiting GFM::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 163059614 bytes to primary GFM file: Homo_sapiens_assembly.1.ht2\n",
      "Wrote 80479924 bytes to secondary GFM file: Homo_sapiens_assembly.2.ht2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from GFM constructor\n",
      "Returning from initFromVector\n",
      "Wrote 1672685819 bytes to primary GFM file: Homo_sapiens_assembly.5.ht2\n",
      "Wrote 80695600 bytes to secondary GFM file: Homo_sapiens_assembly.6.ht2\n",
      "Re-opening _in5 and _in5 as input streams\n",
      "Returning from HGFM constructor\n",
      "Headers:\n",
      "    len: 321919676\n",
      "    gbwtLen: 321919677\n",
      "    nodes: 321919677\n",
      "    sz: 80479919\n",
      "    gbwtSz: 80479920\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 0\n",
      "    eftabSz: 0\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 20119980\n",
      "    offsSz: 80479920\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideGbwtSz: 48\n",
      "    sideGbwtLen: 192\n",
      "    numSides: 1676665\n",
      "    numLines: 1676665\n",
      "    gbwtTotLen: 107306560\n",
      "    gbwtTotSz: 107306560\n",
      "    reverse: 0\n",
      "    linearFM: Yes\n",
      "Total time for call to driver() for forward index: 00:21:23\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "!gunzip Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
    "\n",
    "!hisat2-build -p 4 Homo_sapiens.GRCh38.cdna.all.fa Homo_sapiens_assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "7URvaJ_BmyBv"
   },
   "outputs": [],
   "source": [
    "!mkdir -p \"data\"\n",
    "!mkdir -p \"analysis\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "MD6tawg_lhby",
    "outputId": "ac1b6c92-a001-4733-d80b-ec783b866a9c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sat Mar 27 15:05:46 UTC 2021\n",
      "Sat Mar 27 15:11:30 UTC 2021\n",
      "hisat2.log  reads.raw.bam\n",
      "Sat Mar 27 15:11:30 UTC 2021\n",
      "removed 'analysis/hisat2/reads.raw.bam'\n",
      "Sat Mar 27 15:11:31 UTC 2021\n",
      "Sat Mar 27 15:11:31 UTC 2021\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v2.0.1\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           o reads.bam                                      ||\n",
      "||                                                                            ||\n",
      "||             Output file : reads.fc.txt                                     ||\n",
      "||                 Summary : reads.fc.txt.summary                             ||\n",
      "||              Annotation : Homo_sapiens.GRCh38.100.gtf (GTF)                ||\n",
      "||      Dir for temp files : analysis/featureCounts                           ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 1                                                ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file Homo_sapiens.GRCh38.100.gtf ...                       ||\n",
      "||    Features : 1378275                                                      ||\n",
      "||    Meta-features : 60683                                                   ||\n",
      "||    Chromosomes/contigs : 47                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file reads.bam...                                              ||\n",
      "||    Strand specific : reversely stranded                                    ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Total alignments : 0                                                    ||\n",
      "||    Successfully assigned alignments : 0                                    ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "|| Write the final count table.                                               ||\n",
      "|| Write the read assignment summary.                                         ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"analysis/featureCounts/  ||\n",
      "|| reads.fc.txt.summary\"                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "Sat Mar 27 15:11:55 UTC 2021\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p \"analysis/hisat2/\"\n",
    "!date\n",
    "!hisat2 -p 4 --new-summary -x  Homo_sapiens_assembly -U \"14.1.fastq\"  \"14.2.fastq\" 2> \"analysis/hisat2/hisat2.log\" \\\n",
    "                                                                 | samtools view -b - > \"analysis/hisat2/reads.raw.bam\"\n",
    "!date\n",
    "\n",
    "!ls \"analysis/hisat2/\"\n",
    "#!cat $OUTDIR/$TAG.log\n",
    "\n",
    "##### Step 4: post-processing of alignment\n",
    "\n",
    "# post-processing the alignments\n",
    "\n",
    "!date\n",
    "!samtools sort -@ 4 -O bam \"analysis/hisat2/reads.raw.bam\" > \"analysis/hisat2/reads.bam\" && \\\n",
    "  samtools index \"analysis/hisat2/reads.bam\" && \\\n",
    "  rm -v \"analysis/hisat2/reads.raw.bam\"\n",
    "!date\n",
    " \n",
    "##### Step 7: counting reads\n",
    "\n",
    "!mkdir -p \"analysis/featureCounts/\"\n",
    "!date\n",
    "!featureCounts -a \"Homo_sapiens.GRCh38.100.gtf\" -s 2 -o \"analysis/featureCounts/reads.fc.txt\" \"analysis/hisat2/reads.bam\" |& tee \"analysis/featureCounts/reads.log\"\n",
    "!date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 107
    },
    "id": "20GwODvLhkPZ",
    "outputId": "a69c4260-d629-4b96-f03b-e08ce4c843c1"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "updating: analysis/featureCounts/ (stored 0%)\n",
      "updating: analysis/featureCounts/reads.fc.txt.summary (deflated 54%)\n",
      "updating: analysis/featureCounts/reads.log (deflated 80%)\n",
      "updating: analysis/featureCounts/reads.fc.txt (deflated 85%)\n"
     ]
    },
    {
     "data": {
      "application/javascript": [
       "\n",
       "    async function download(id, filename, size) {\n",
       "      if (!google.colab.kernel.accessAllowed) {\n",
       "        return;\n",
       "      }\n",
       "      const div = document.createElement('div');\n",
       "      const label = document.createElement('label');\n",
       "      label.textContent = `Downloading \"${filename}\": `;\n",
       "      div.appendChild(label);\n",
       "      const progress = document.createElement('progress');\n",
       "      progress.max = size;\n",
       "      div.appendChild(progress);\n",
       "      document.body.appendChild(div);\n",
       "\n",
       "      const buffers = [];\n",
       "      let downloaded = 0;\n",
       "\n",
       "      const channel = await google.colab.kernel.comms.open(id);\n",
       "      // Send a message to notify the kernel that we're ready.\n",
       "      channel.send({})\n",
       "\n",
       "      for await (const message of channel.messages) {\n",
       "        // Send a message to notify the kernel that we're ready.\n",
       "        channel.send({})\n",
       "        if (message.buffers) {\n",
       "          for (const buffer of message.buffers) {\n",
       "            buffers.push(buffer);\n",
       "            downloaded += buffer.byteLength;\n",
       "            progress.value = downloaded;\n",
       "          }\n",
       "        }\n",
       "      }\n",
       "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
       "      const a = document.createElement('a');\n",
       "      a.href = window.URL.createObjectURL(blob);\n",
       "      a.download = filename;\n",
       "      div.appendChild(a);\n",
       "      a.click();\n",
       "      div.remove();\n",
       "    }\n",
       "  "
      ],
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {
      "tags": []
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": [
       "download(\"download_161e3fc6-b991-4f8f-ba16-d828617e44e3\", \"featureCounts.zip\", 4917859)"
      ],
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {
      "tags": []
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "from google.colab import files\n",
    "!zip -r featureCounts.zip analysis/featureCounts/\n",
    "files.download('featureCounts.zip') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "ddEdjYZClcW6",
    "outputId": "9d9f376e-934f-45d4-9fa0-5c73bb144fdf"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-03-27 16:54:26--  ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/old/minikraken2_v1_8GB_201904.tgz\n",
      "           => ‘minikraken2_v1_8GB_201904.tgz’\n",
      "Resolving ftp.ccb.jhu.edu (ftp.ccb.jhu.edu)... 128.220.174.63\n",
      "Connecting to ftp.ccb.jhu.edu (ftp.ccb.jhu.edu)|128.220.174.63|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/data/kraken2_dbs/old ... done.\n",
      "==> SIZE minikraken2_v1_8GB_201904.tgz ... 6016594204\n",
      "==> PASV ... done.    ==> RETR minikraken2_v1_8GB_201904.tgz ... done.\n",
      "Length: 6016594204 (5.6G) (unauthoritative)\n",
      "\n",
      "minikraken2_v1_8GB_ 100%[===================>]   5.60G  44.9MB/s    in 2m 35s  \n",
      "\n",
      "2021-03-27 16:57:02 (37.1 MB/s) - ‘minikraken2_v1_8GB_201904.tgz’ saved [6016594204]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/old/minikraken2_v1_8GB_201904.tgz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "NxfgiaSanlgF",
    "outputId": "ee067a8d-38ee-43f5-d888-c55e33978ce1"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14.1.fastq\t\t\t   Homo_sapiens_assembly.4.ht2\n",
      "14.2.fastq\t\t\t   Homo_sapiens_assembly.5.ht2\n",
      "analysis\t\t\t   Homo_sapiens_assembly.6.ht2\n",
      "cseqs09_1.fq\t\t\t   Homo_sapiens_assembly.7.ht2\n",
      "cseqs09_2.fq\t\t\t   Homo_sapiens_assembly.8.ht2\n",
      "cseqs099_1.fq\t\t\t   KRAKEN\n",
      "cseqs099_2.fq\t\t\t   kraken2\n",
      "cseqs1_1.fq\t\t\t   Miniconda3-py37_4.8.2-Linux-x86_64.sh\n",
      "cseqs1_2.fq\t\t\t   minikraken2_v1_8GB_201904.tgz\n",
      "cseqs_1.fq\t\t\t   mm10_Gencode_VM18.bed\n",
      "cseqs_2.fq\t\t\t   mm10.HouseKeepingGenes.bed\n",
      "data\t\t\t\t   mm10_rRNA.bed\n",
      "featureCounts.zip\t\t   refFlat.txt\n",
      "gencode.vM20.transcripts.fa.gz\t   REPORT_NCBI_VIRAL_08.report\n",
      "GRCm38.primary_assembly.1.ht2\t   REPORT_NCBI_VIRAL099.report\n",
      "GRCm38.primary_assembly.2.ht2\t   REPORT_NCBI_VIRAL09.report\n",
      "GRCm38.primary_assembly.3.ht2\t   REPORT_NCBI_VIRAL1.report\n",
      "GRCm38.primary_assembly.4.ht2\t   REPORT_NCBI_VIRAL.report\n",
      "GRCm38.primary_assembly.7.ht2\t   sample_data\n",
      "GRCm38.primary_assembly.8.ht2\t   steam.out\n",
      "GRCm38.primary_assembly.genome.fa  viral.1.1.genomic.fna\n",
      "Homo_sapiens_assembly.1.ht2\t   viral.2.1.genomic.fna\n",
      "Homo_sapiens_assembly.2.ht2\t   viral.3.1.genomic.fna\n",
      "Homo_sapiens_assembly.3.ht2\t   ViralAttempt3\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "3MEKHfGfldS7",
    "outputId": "fdb72b1c-e4c2-418e-c3c1-c4a917c88a41"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14.1.fastq\t\t\t       minikraken2_v1_8GB_201904.tar\n",
      "14.2.fastq\t\t\t       mm10_Gencode_VM18.bed\n",
      "analysis\t\t\t       mm10.HouseKeepingGenes.bed\n",
      "cseqs09_1.fq\t\t\t       mm10_rRNA.bed\n",
      "cseqs09_2.fq\t\t\t       refFlat.txt\n",
      "cseqs099_1.fq\t\t\t       REPORT_NCBI_VIRAL_08.report\n",
      "cseqs099_2.fq\t\t\t       REPORT_NCBI_VIRAL099.report\n",
      "cseqs1_1.fq\t\t\t       REPORT_NCBI_VIRAL09.report\n",
      "cseqs1_2.fq\t\t\t       REPORT_NCBI_VIRAL1.report\n",
      "cseqs_1.fq\t\t\t       REPORT_NCBI_VIRAL.report\n",
      "cseqs_2.fq\t\t\t       sample_data\n",
      "data\t\t\t\t       steam.out\n",
      "featureCounts.zip\t\t       viral.1.1.genomic.fna\n",
      "gencode.vM20.transcripts.fa.gz\t       viral.2.1.genomic.fna\n",
      "KRAKEN\t\t\t\t       viral.3.1.genomic.fna\n",
      "kraken2\t\t\t\t       ViralAttempt3\n",
      "Miniconda3-py37_4.8.2-Linux-x86_64.sh\n"
     ]
    }
   ],
   "source": [
    "!gunzip minikraken2_v1_8GB_201904.tgz\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "m6hlKWkdmltc",
    "outputId": "a4c55cc9-ff11-4c90-e421-6f64ccff4752"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "minikraken2_v1_8GB/\n",
      "minikraken2_v1_8GB/taxo.k2d\n",
      "minikraken2_v1_8GB/database150mers.kmer_distrib\n",
      "minikraken2_v1_8GB/opts.k2d\n",
      "minikraken2_v1_8GB/seqid2taxid.map\n",
      "minikraken2_v1_8GB/database200mers.kmer_distrib\n",
      "minikraken2_v1_8GB/database100mers.kmer_distrib\n",
      "minikraken2_v1_8GB/hash.k2d\n",
      "14.1.fastq\t\t\t       minikraken2_v1_8GB\n",
      "14.2.fastq\t\t\t       minikraken2_v1_8GB_201904.tar\n",
      "analysis\t\t\t       mm10_Gencode_VM18.bed\n",
      "cseqs09_1.fq\t\t\t       mm10.HouseKeepingGenes.bed\n",
      "cseqs09_2.fq\t\t\t       mm10_rRNA.bed\n",
      "cseqs099_1.fq\t\t\t       refFlat.txt\n",
      "cseqs099_2.fq\t\t\t       REPORT_NCBI_VIRAL_08.report\n",
      "cseqs1_1.fq\t\t\t       REPORT_NCBI_VIRAL099.report\n",
      "cseqs1_2.fq\t\t\t       REPORT_NCBI_VIRAL09.report\n",
      "cseqs_1.fq\t\t\t       REPORT_NCBI_VIRAL1.report\n",
      "cseqs_2.fq\t\t\t       REPORT_NCBI_VIRAL.report\n",
      "data\t\t\t\t       sample_data\n",
      "featureCounts.zip\t\t       steam.out\n",
      "gencode.vM20.transcripts.fa.gz\t       viral.1.1.genomic.fna\n",
      "KRAKEN\t\t\t\t       viral.2.1.genomic.fna\n",
      "kraken2\t\t\t\t       viral.3.1.genomic.fna\n",
      "Miniconda3-py37_4.8.2-Linux-x86_64.sh  ViralAttempt3\n"
     ]
    }
   ],
   "source": [
    "!tar -xvf minikraken2_v1_8GB_201904.tar\n",
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "n8qsgpxynYVH",
    "outputId": "5cc63d3c-f00c-4829-9bf6-a0c9d6fa68ad"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information...tcmalloc: large alloc 8000004096 bytes == 0x556b77080000 @  0x7f5727878887 0x556b75d4008d 0x556b75d402d8 0x556b75d2a403 0x7f57266e6bf7 0x556b75d2aeea\n",
      " done.\n",
      "Processed 40000 sequences (5261008 bp) ...classify: malformed FASTQ file (exp. '@', saw \"SimSeq_3/1\t0\tENST00000455382.2\t199\t1\t100M\t*\t0\t0\tGTGTCCCCATCTCTAATCACTTATACTTCTATTGGTACAGACAAATCTTGGGGCAGAAAGTCGAGTTTCTGGTTTCCTTTTATAATAATGAAATCTCAGA\t~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYT:Z:UU\tNH:i:2\"), aborting\n"
     ]
    }
   ],
   "source": [
    "!kraken2/KRAKEN/kraken2 --threads 1 --db minikraken2_v1_8GB --paired --classified-out cseqs1#.fq 14.1.fastq  14.2.fastq  --confidence 0.9 --report REPORT_NCBI_VIRAL1.report  -output steam.out"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [],
   "name": "SeaMosters_KRAKEN.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
